index.js

```diff
@@ -145,17 +145,19 @@ function _render (vnode, options, hooks) {
     return vnode.map(function (vnode) { return _render(vnode, options, hooks) }).join('')
   }
 
+
   // component
   if (typeof vnode.tag === 'object' && vnode.tag.view) {
-    var component = vnode.tag
-    vnode.state = copy(component)
-
+    vnode.state = copy(vnode.tag)
     oninit(vnode)
-    var result = _render(component.view.call(vnode.state, vnode), options, hooks)
     onremove(vnode, hooks)
-    return result
+    return _render(vnode.tag.view.call(vnode.state, vnode), options, hooks)
   }
 
+  oninit(vnode)
+  onremove(vnode, hooks)
```

Comment:
I know it looks strange to put them here, but that's what Mithril does. Any kind of `vdom` node can have hooks on its attribute object.